cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -582,15 +587,12 @@ addLovelace a b = assumeBound $ CC.Common.addLovelace a b
 --   with it atomically. There to support potential multithreaded use.
 -----------------------------------------------------------------------------------------
 
--- | Set of available tx outputs (for new transactions).
-availableFunds :: TVar.TVar AvailableFunds
-availableFunds = unsafePerformIO $ TVar.newTVarIO Set.empty
-
 -- | Add to the available funds.
 addToAvailableFunds
-  :: TxDetails
+  :: TVar.TVar AvailableFunds
+  -> TxDetails
   -> IO ()
-addToAvailableFunds txDetails = do
+addToAvailableFunds availableFunds txDetails = do
```

Comment:
This function could cease to exist completely, if we follow the suggested way of purifying the available funds (money laundering, huh? : -).